openmc/plotter.py

```diff
@@ -540,3 +602,305 @@ def _calculate_xs_elem_mat(this, types, temperature=294., cross_sections=None,
                                   xs[name][line](energy_grid))
 
     return energy_grid, data
+
+
+def calculate_mgxs(this, types, orders=None, temperature=294.,
+                   cross_sections=None, ce_cross_sections=None,
+                   enrichment=None):
+    """Calculates continuous-energy cross sections of a requested type.
+
+    If the data for the nuclide or macroscopic object in the library is
+    represented as angle-dependent data then this method will return the
+    geometric average cross section over all angles.
+
+    Parameters
+    ----------
+    this : openmc.Element, openmc.Nuclide, or openmc.Material
```

Comment:
It should include a macroscopic, but not only macroscopic.  You can build an MG model with any of these four types.